### PR TITLE
adding flask support

### PR DESCRIPTION
**Pull request by d-grossman**
Query GPU statistics via flask

### CODE REVIEW

File: gpustat.py [hunk 1]
```diff
@@ -405,14 +410,18 @@ def print_gpustat(json=False, **args):
...
     try:
         gpu_stats = GPUStatCollection.new_query()
     except CalledProcessError:
         sys.stderr.write('Error on calling nvidia-smi\n')
         sys.exit(1)
 
     if json:
-        gpu_stats.print_json(sys.stdout)
+        if args['ret_json']:
+            return gpu_stats.print_json(None)
```
Comment (wookayin) [inline gpustat.py]: Although this line would return the JSON-like object as dict, it doesn't match the name and looks confusing a bit.

File: gpustat.py [hunk 2]
```diff
@@ -434,6 +443,8 @@ def main():
                         help='Print all the information in JSON format')
     parser.add_argument('-v', '--version', action='version',
                         version=('gpustat %s' % __version__))
+    parser.add_argument('--ret-json', action='store_true', default=False,
+                        help='return json for calling as library')
```
Comment (wookayin) [inline gpustat.py]: I don't think this argument is necessary. It looks somewhat misleading; for the sake of getting the object, we can just call the function or parse the result.

File: gpustat.py [hunk 3]
```diff
@@ -262,22 +263,22 @@ def running_processes():
...
-            ))
-            for line in pid_output.split('\n'):
-                if (not line) or 'PID' in line: continue
-                pid, user, comm = line.split()[:3]
-                pid_map[int(pid)] = {
-                    'user' : user,
-                    'comm' : comm
-                }
+        pid_map = {int(e['pid']) : None for e in process_entries if not 'Not Supported' in e['pid']}
+        if os.environ.get('DOCKER_CONTAINER') is  None:
```
Comment (wookayin) [inline gpustat.py]: The existence of docker should be transparent to the code itself, IMHO.

Comment (d-grossman) [inline gpustat.py]: The problem is nvidia-smi returns process ids running on the host.  The container does not necessarily have access to those pids.  When the docker program attempts to introspect the host pids the program exits on bad return. 

I'll paste some debug when I get back into work next week.

File: gpustat.py (hunk 1 above)
Comment (d-grossman) [inline gpustat.py]: I'll refactor/reword this then.

File: gpustat.py (hunk 3 above)
Comment (d-grossman) [reply gpustat.py]: raceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "flask_app.py", line 18, in get_status
    data = gpustat.print_gpustat(**{'show_cmd': False, 'gpuname_width': 16, 'show_user': False, 'show_pid': False, 'json': True, 'no_color': False, 'ret_json': True})
  File "/home/stats/gpustat.py", line 414, in print_gpustat
    gpu_stats = GPUStatCollection.new_query()
  File "/home/stats/gpustat.py", line 246, in new_query
    return GPUStatCollection(gpu_list)
  File "/home/stats/gpustat.py", line 219, in __init__
    self.update_process_information()
  File "/home/stats/gpustat.py", line 310, in update_process_information
    processes = self.running_processes()
  File "/home/stats/gpustat.py", line 274, in running_processes
    ','.join(map(str, pid_map.keys()))
  File "/home/stats/gpustat.py", line 47, in execute_process
    stdout = check_output(command_shell, shell=True).strip()
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command 'ps -o pid,user:16,comm -p 110117,101614' returned non-zero exit status 1

the pids exist on the host, not in the container so nonzero exit script dies.

File: gpustat.py (hunk 2 above)
Comment (d-grossman) [reply gpustat.py]: I made a new function and call that function to give me the json.

**Pull request closed without merging**